cloud/migrate-mst-cloud/schema-then-data.md

```diff
@@ -0,0 +1,352 @@
+    ```
+1.  Use `timescaledb-parallel-copy` to import data into
+    your Cloud database. Set `<num_workers>` to twice the number of CPUs in your
+    database. For example, if you have 4 CPUs, `<num_workers>` should be `8`.
+    ```bash
+    timescaledb-parallel-copy \
+    --connection "host=<cloud_host> \
+    user=tsdbadmin password=<cloud_password> \
+    port=<cloud_port> \
+    sslmode="require" \
+    --db-name tsdb \
+    --table <table_name> \
+    --file <file_name>.csv \
+    --workers <num_workers> \
+    --reporting-period 30s
+    ```
+    Repeat for each table and hypertable you want to migrate.
+
+</procedure>
+
+<procedure>
+
+### Restoring data into Timescale Cloud with COPY
+1.  Connect to your Timescale Cloud database:
+    ```sql
+    psql "postgres://tsdbadmin:<cloud_password>@<cloud_host>:<cloud_port>/tsdb?sslmode=require"
+    ```
+1.  Restore the data to your Timescale Cloud database:
+    ```sql
+    COPY <table-name> FROM '<table-name>.csv' WITH (FORMAT CSV);
+    ```
+    Repeat for each table and hypertable you want to migrate.
+
+</procedure>
+
+## Migrate schema post-data
+When you have migrated table and hypertable data, migrate your PostgreSQL schema
+post-data. This includes information about constraints.
+
+<procedure>
+
+### Migrating schema post-data
+1.  At the command prompt, dump the schema post-data from Managed Service for
+    TimescaleDB into a `.bak` file, using your Managed Service for TimescaleDB
+    connection details. Exclude Timescale-specific schemas. When you are
+    prompted for a password, use your Managed Service for TimescaleDB
+    credentials:
+    ```bash
+    pg_dump -U tsdbadmin -W \
+    -h <mst_host> -p <mst_port> -Fc -v \
+    --section=post-data --exclude-schema="_timescaledb*" \
+    -f <file-name-post-data>.bak defaultdb
+    ```
+1.  Restore the dumped schema post-data from the `.bak` file into your Timescale
+    Cloud database, using your Timescale Cloud connection details. To avoid
+    permissions errors, include the `--no-owner` flag:
+    ```bash
+    pg_restore -U tsdbadmin -W \
+    -h <cloud_host> -p <cloud_port> --no-owner -Fc \
+    -v -d tsdb <file-name-post-data>.bak
+    ```
+
+</procedure>
+
+### Troubleshooting
+If you see these errors during the migration process, you can safely ignore
+them. The migration still occurs successfully.
+
+```bash
+pg_restore: error: could not execute query: ERROR:  relation "<relation_name>" already exists
+```
+
+```bash
+pg_restore: error: could not execute query: ERROR:  trigger "ts_insert_blocker" for relation "<relation_name>" already exists
+```
+
+## Recreate continuous aggregates
+By default, continuous aggregates aren't migrated when you transfer your schema
+and data separately. Restore them by recreating the continuous aggregate
+definitions and recomputing the results on your Cloud database.
+
+<procedure>
+
+### Recreating continuous aggregates
```

Comment:
should we add the warning about migrating CAGGs here as well?

Reply:
Adding a modified (shorter) version. Hopefully no user will get this far if CAGGs are a problem for them. To avoid that, I'm adding the full warning to the top of the page for anyone who gets here without passing the index page.

Reply:
it's probably mostly useful for users who first want to read through the docs before actually migrating.